cmd/snap-confine/seccomp-support.c

```diff
@@ -109,13 +109,38 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
-bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag)
-{
+bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag) {
 	debug("loading bpf program for security tag %s", security_tag);
 
 	char profile_path[PATH_MAX] = { 0 };
-	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin",
+
+        struct sock_fprog prog_allow = { 0 };
+	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin.allow",
+			 filter_profile_dir, security_tag);
+        if (!sc_load_seccomp_profile_path(profile_path, &prog_allow)) {
+           return false;
+        }
+
+        struct sock_fprog prog_deny  = { 0 };
+	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin.deny",
 			 filter_profile_dir, security_tag);
+        if (!sc_load_seccomp_profile_path(profile_path, &prog_deny)) {
+           return false;
+        }
+
+        sc_apply_seccomp_filter(&prog_deny);
+        sc_apply_seccomp_filter(&prog_allow);
+
+        // XXX: this is not ideal, allocation happens in
```

Comment:
Perhaps it would be worth defining a new function `void sc_cleanup_seccomp_profile(struct sock_fprog *prog)` to do this?